### DISCUSSION TITLE
v1.1.42 JPEG XL support - Happy Lunar New Year

### DISCUSSION THREAD

**Discussion by sylikc** [sylikc/jpegview#117] [General]
## 1.1.42

Happy Lunar New Year!!!  I wanted to make a special thanks to [qbnu](https://github.com/qbnu) for adding JPEG XL support to JPEGView!

Lots of new features to highlight along with new INI settings!

### New features:
- **JPEG XL support, with animation**
  (NOTE: Microsoft Visual C++ Runtime vcruntime140.dll required when opening
  JPEG XL files.  Most Windows 10+ systems have this installed already.
  Otherwise, download it for free from Microsoft.)
  PR #99, thanks [qbnu](https://github.com/qbnu)
- Add a new command `Hide window title bar`
  Default shortcut key: `SHIFT+F11`
  NOTE: in this mode, you cannot resize the window (although full screen is still supported)
- Add a new mouse gesture to move the JPEGView window at any time: `ALT+LeftClick`
  This allows moving the window without directly clicking on the title bar (or if title bar is hidden).
- Add a new command `Always on Top`
  Default shortcut key: `SHIFT+F12`
  This toggles the JPEGView window to display above all other windows, even when not in focus
- Add new command "Copy file path"
  Default shortcut key: "CTRL+SHIFT+C"
  This copies file path of the image to clipboard
  PR #62, thanks [Trevor Kropp](https://github.com/kropptrevor)

### Changed behavior:
- Animated formats: frame durations <= 10ms are set to 100ms
  (Matches behavior of most web browers)
  PR #108, thanks [qbnu](https://github.com/qbnu)

### Other changes:
- New INI file setting: `ShowFullPathInTitle`
  Show full file path and name in the window title
  (shows only file name by default)
- New INI file setting: `DefaultSelectionMode`
  Allows for disabling the default selection mode behavior when clicking on an image in JPEGView.
  Set to false to use selection mode only when CTRL is pressed, to prevent unintended cropping/zooming.
  (`CTRL+LeftClick` always results in selection, regardless of image zoom)
- New `KeyMap.txt` entry to allow a keybind for `IDM_CHANGESIZE`
  This allows binding a key to the "Transform image > Change size..." popup dialog

<hr /><em>This discussion was created from the release <a href='https://github.com/sylikc/jpegview/releases/tag/v1.1.42'>v1.1.42 JPEG XL support - Happy Lunar New Year</a>.</em>

**Comment by c-sanchez** [comment]
You are a f*cking genius, really thank you very much for updating JPEGView.
Now it can be said without a doubt that your fork is a substantial improvement of the original JPEGView, which was already really great, but now it's even better.

**Reply by slovuj** [reply]
_New INI file setting: ShowFullPathInTitle
Show full file path and name in the window title (shows only file name by default)_

Is this necessary? I've had this for ages in my INI:

; The elements to show when showing the file name.                               
; Possible elements:
; %filename% : File name
**`; %filepath% : File name and path`**
; %index%    : Index of image in folder, e.g. [1/12]
; %zoom%     : Current zoom factor
; %size%     : Size of image in pixels (w x h)
; %filesize% : Size of image on disk
FileNameFormat=%size%  %zoom%  (%filesize%) %index% **`%filepath%`**

**Reply by sylikc** [reply]
So, before adding the feature, I looked into this as well.  This specific setting is for the text that shows in the upper left of the window, rather than the title bar.

Not everyone has the ShowFileInfo set to true or ever cause that to show.  Or feels like drawing over the image is annoying, etc etc.  So the new INI setting is to show it in the title bar.

**Reply by slovuj** [reply]
Oh, it's about the window title bar. Sorry, I wasn't paying attention.

**Comment by edihooo** [comment]
an advice, ALT+LeftClick, thank you for adding this finally, now how about set an option to avoid ALT, but drag to move window directly? since it's the most efficient way to rearrange windows(dont' have to aim at the title bar). Also, with this people can hide title bar without worries to make the looking more simple and clear. thank you.

**Comment by MSAfotografia** [comment]
Hello, is it possible to add the option of not confirming to delete photos? I select more than 1,000 photos, leaving only 300 on average, and every time I delete a photo I have to confirm that I have to delete it. If this is possible, I would be very grateful if someone could tell me how to do it. 
![pt br](https://github.com/user-attachments/assets/8a9b9ccf-9776-43ec-b39f-a08ec476893e)

**Reply by qbnu** [reply]
Set `DeleteConfirmation=Never` in JPEGView.ini. To delete with the Del key follow https://github.com/sylikc/jpegview/issues/309
